languages/cpp2/src/test/java/de/jplag/cpp2/LanguageTest.java

```diff
@@ -0,0 +1,206 @@
+package de.jplag.cpp2;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.opentest4j.TestAbortedException;
+
+import de.jplag.ParsingException;
+import de.jplag.SharedTokenType;
+import de.jplag.Token;
+import de.jplag.TokenPrinter;
+import de.jplag.TokenType;
+
+class LanguageTest {
+
+    @ParameterizedTest
+    @ValueSource(strings = {"i = 10", "i += 10", "i -= 10", "i += 10", "i /= 10", "i %= 10", "i >>= 10", "i <<= 10", "i &= 10", "i ^= 10", "i |= 10",
+            "i++", "i--", "++i", "--i",})
+    void testAssign(String expr, @TempDir Path path) {
+        String function = """
+                void f(int i) {
+                    %s;
+                }
+                """.formatted(expr);
+        List<Token> all = extractFromString(path, function).tokens();
+        List<Token> assignTokens = all.stream().filter(token -> token.getType() == CPPTokenType.ASSIGN).toList();
+        assertEquals(1, assignTokens.size());
+    }
+
+    @Test
+    void testFunctionCall(@TempDir Path path) {
+        TokenResult result = extractFromString(path, """
```

Comment:
These multi-line strings could be constants. What do you think, does it improve readability or just make it more difficult?

Reply:
I think having them directly at the test case is more readable.

I'm not sure if it's worth the work, but introducing a custom annotation for JUnit that takes a String and directly injects the result as parameters of the test method would be doable too (but there might be some more design aspects to consider then).

Reply:
Then let's leave it as is, as the strings are so central to the test case (as opposed to e.g. a file path), I had no strong opinion one way or the other.